eth/blockwatch/block_watcher.go

```diff
@@ -146,6 +146,25 @@ func (w *Watcher) InspectRetainedBlocks() ([]*MiniHeader, error) {
 	return w.stack.Inspect()
 }
 
+func (w *Watcher) syncToLatestBlock() error {
+	lastBlock, err := w.client.HeaderByNumber(nil)
+	if err != nil {
+		return err
+	}
+
+	for {
+		latestHeader, err := w.stack.Peek()
```

Comment:
Currently, we call Peek() once for each loop iteration. What about calling Peek() once, calculating the # of headers to fetch and then running the loop that many times? Avoids the extra DB reads and a tad bit more explicit when the loop will end.

Reply:
fixed in https://github.com/livepeer/go-livepeer/pull/1729/commits/511c85802c591398ed2ba931af5013a70cea502a